topdown/glob.go

```diff
@@ -33,7 +33,13 @@ func builtinGlobMatch(a, b, c ast.Value) (ast.Value, error) {
 		return nil, err
 	}
 
-	id := fmt.Sprintf("%s-%v", pattern, delimiters)
+	builder := strings.Builder{}
+	builder.WriteString(string(pattern))
```

Comment:
Would it make a difference to use the pattern length and the number of delimiters to grow the builder to the required size upfront?

Reply:
Yeah, I tried a number of different approaches, including just using bare strings and the `+` operator directly. Nothing really seemed to move the needle, so I think the real benefit here is just eliminating the constant factor `Sprintf()` overhead. That and not paying for string quoting by doing a direct typecast.